### PR TITLE
2269 publish events from wmi

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #2269 .

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] ~Was the CHANGELOG.md updated to reflect the changes?~
* [x] ~Was the documentation framework updated to reflect the changes?~
* [x] ~Have you checked that you haven't introduced any duplicate code?~

## Testing Checklist

* [x] ~Added relevant unit tests?~
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/infection_monkey/exploit/wmiexec.py [hunk 1]
```diff
@@ -21,10 +27,15 @@
 
 logger = logging.getLogger(__name__)
 
+WMI_EXPLOITER_TAG = "wmi-expliter"
```
Comment (shreyamalviya) [inline monkey/infection_monkey/exploit/wmiexec.py:30]:
```suggestion
WMI_EXPLOITER_TAG = "wmi-exploiter"
```

**Review by shreyamalviya — CHANGES_REQUESTED** [review]
We need to change when the propagation event is published based on our discussion.

File: monkey/infection_monkey/exploit/wmiexec.py [hunk 2]
```diff
@@ -113,33 +133,30 @@ def _exploit_host(self) -> ExploiterResultData:
                 } + build_monkey_commandline(self.servers, self.current_depth + 1)
 
             # execute the remote monkey
+            propagation_timestamp = time()
```
Comment (mssalvatore) [inline monkey/infection_monkey/exploit/wmiexec.py:136]: As discussed in our daily sync, propagation begins when the file is copied.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/infection_monkey/exploit/wmiexec.py (hunk 2 above)
Comment (ilija-lazoroski) [reply monkey/infection_monkey/exploit/wmiexec.py]: Changed

Comment (cakekoa) [inline monkey/infection_monkey/exploit/wmiexec.py]: Based on our discussion, we should move this timestamp to where the agent binary is copied to the victim machine.

File: monkey/infection_monkey/exploit/wmiexec.py [hunk 3]
```diff
@@ -55,26 +67,34 @@ def _exploit_host(self) -> ExploiterResultData:
...
                     f"Unknown WMI connection error to {self.host}: "
                     f"{exc} {traceback.format_exc()}"
                 )
+                logger.debug(error_message)
+                self._publish_exploitation_event(timestamp, False, error_message=error_message)
                 return self.exploit_result
 
             self.report_login_attempt(True, user, password, lm_hash, ntlm_hash)
             self.exploit_result.exploitation_success = True
+            self._publish_exploitation_event(timestamp, False, error_message=error_message)
```
Comment (cakekoa) [inline monkey/infection_monkey/exploit/wmiexec.py:97]: Shouldn't this be `True`?

**Review by shreyamalviya — APPROVED** [review]

File: monkey/infection_monkey/exploit/wmiexec.py (hunk 3 above)
Comment (shreyamalviya) [reply monkey/infection_monkey/exploit/wmiexec.py:97]: Good catch

**Review by cakekoa — APPROVED** [review]

**Pull request merged**